lib/IDE/Refactoring.cpp

```diff
@@ -4501,22 +4507,14 @@ class ClassifiedBlock {
     }
 
     StringRef Name = FromCondition.BindPattern->getBoundName().str();
-    if (Name.empty())
-      return;
-
-    auto Res = BoundNames.try_emplace(FromCondition.Subject, Name);
-    if (Res.second)
+    VarDecl *SingleVar = FromCondition.BindPattern->getSingleVar();
+    if (Name.empty() || !SingleVar)
       return;
```

Comment:
Just noticed now, but what if it binds multiple vars? For example:

```
func simple(_ completion: ((String, String)?, Error?) -> Void) {}

simple { strs, err in
  guard let (str1, str2) = strs else {
    return
  }
  print(str1, str2)
}

```

Reply:
Good point, this currently generates invalid code:
```
let strs = try await simple()
print(str1, str2)
```

Not sure how common this would be, but probably worth fixing (in another PR since the underlying problem is unrelated to this change).